pkg/download/extract.go

```diff
@@ -94,6 +95,21 @@ func openArchive(archivePath string, password string) (*archiveInfo, error) {
 	}, nil
 }
 
+// createExtractionHandler creates a handler function for extracting files with progress tracking
+func createExtractionHandler(destDir string, totalFiles int, extractedFiles *atomic.Int32, progressCallback ExtractProgressCallback) func(ctx context.Context, fileInfo archives.FileInfo) error {
+	return func(ctx context.Context, fileInfo archives.FileInfo) error {
```

Comment:
I don't think there's a need to have `extractedFiles` as a parameter to the function.
I think it can be defined here, before the return of this function.